src/main/java/com/walmartlabs/x12/X12Parser.java

```diff
@@ -53,9 +64,28 @@
             return segments;
         } else {
             // if there is only one line in the source data
-            // we should attempt to use the segment separator passed in
-            // and see if we can split up this source data
-            return splitSourceDataIntoSegments(sourceData, "\\" + DEFAULT_SEGMENT_SEPARATOR);
+            // we should attempt to use the default segment separator
+            segments = splitSourceDataIntoSegments(sourceData, "\\" + DEFAULT_SEGMENT_SEPARATOR);
+            if (segments != null && segments.size() > 1) {
```

Comment:
can we use
`segments != null && !segments.isEmpty()` or
`CollectionUtils.isEmpty(segments)`